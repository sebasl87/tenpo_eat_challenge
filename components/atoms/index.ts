export { default as RestaurantsLayout } from './RestaurantsLayout';
