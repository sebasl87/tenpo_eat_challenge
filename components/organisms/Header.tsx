import React from 'react'
import { Image, Text, View } from 'react-native';
import { styles } from '../../styles/styles';

function Header() {
    return (
        <>
            <View style={styles.header}>
                <Image
                    source={require("../../assets/imgs/user.png")}
                    style={styles.userImg}
                />
                <Image
                    source={require("../../assets/icons/search.png")}
                    style={styles.searchImg}
                />
            </View>
            <View style={styles.header}>
                <View style={{ flexDirection: "column" }}>
                    <Image
                        source={require("../../assets/imgs/logo.png")}
                        style={styles.logoImg}
                    />
                    <Text>DELIVER APP</Text>
                </View>
                <View style={{ flexDirection: "row", marginBottom: -16 }}>
                    <Image
                        source={require("../../assets/imgs/hand.png")}
                        style={styles.handImg}
                    />
                    <View style={{ top: "100%", height: 6 }}>

                        <Image
                            source={require("../../assets/icons/ellipse.png")}
                            style={styles.ellipseImg}
                        />
                    </View>
                </View>
            </View>
        </>
    );
}

export default Header