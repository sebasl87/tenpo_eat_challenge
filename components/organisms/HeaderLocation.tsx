import React from 'react'
import { Image, Text, TextInput, View } from 'react-native';
import { styles } from '../../styles/styles';

function Header() {
    return (
        <>
            <View style={styles.headerLocation}>
                <View style={{ marginBottom: 32, display: "flex", flexDirection: "row" }}>
                    <Image
                        source={require("../../assets/icons/location.png")}
                        style={{ height: 24, width: 24, marginRight: 6, paddingTop: 30 }}
                    />
                    <Text style={styles.fontTitle}>Agregar dirección de la entrega</Text>
                </View>
            </View>
            <TextInput placeholder='Escribe tu dirección' style={styles.inputLocation} />
        </>
    );
}

export default Header