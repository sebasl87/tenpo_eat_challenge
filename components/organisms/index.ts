export { default as Header } from "./Header";
export { default as HeaderLocation } from "./HeaderLocation";
