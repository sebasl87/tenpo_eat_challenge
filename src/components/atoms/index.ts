import AddressLayout from './AddressLayout';
export { default as RestaurantsLayout } from './RestaurantsLayout';
export { default as CategoriesLayout } from './CategoriesLayout';
export { default as HeaderFavorites } from './HeaderFavorites';
export { default as AddressLayout } from './AddressLayout';
