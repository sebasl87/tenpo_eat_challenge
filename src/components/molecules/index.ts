export { default as DetailCard } from './DetailCard';
export { default as DiscountCircle } from './DiscountCircle';
export { default as LogoRounded } from './LogoRounded';
export { default as DetailFavorites } from './DetailFavorites';
