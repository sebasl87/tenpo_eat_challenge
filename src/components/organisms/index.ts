export { default as Header } from "./Header";
export { default as HeaderLocation } from "./HeaderLocation";
export { default as ContainerMapView } from "./ContainerMapView";
export { default as RestaurantRow } from "./RestaurantRow";
export { default as ContainerAddressData } from "./ContainerAddressData";
export { default as CategoriesRow } from "./CategoriesRow";
export { default as FavoritesRow } from "./FavoritesRow";
export { default as HomeLocation } from './HomeLocation';
