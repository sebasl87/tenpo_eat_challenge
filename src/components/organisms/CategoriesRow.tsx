import React from 'react'
import { View, ScrollView, Text } from 'react-native';
import styled from 'styled-components/native';
import { CategoriesLayout } from '../atoms'

export interface ICategories { }

const Title = styled.Text`
  color: #333333;
  font-size: 18px;
  margin-bottom: 6px;
  margin-top: 32px;
`;

const linksData = [
    {
        uri: require("../../../assets/imgs/hamburguer.png"),
        category: "HAMBURGUESAS"
    },
    {
        uri: require("../../../assets/imgs/italian.png"),
        category: "ITALIANA"
    },
    {
        uri: require("../../../assets/imgs/pizza.png"),
        category: "PIZZAS"
    },

]

const CategoriesRow: React.FC<ICategories> = () => (
    <>

        <Title style={{ fontFamily: "Gotham-Bold" }}>CATEGORIAS</Title>
        <ScrollView horizontal={true}>
            <View style={{ flexDirection: "row" }}>
                {linksData.map((item, i) => (
                    <View key={i} style={{ marginRight: 6 }}>

                        <CategoriesLayout image={item.uri}>{item.category}</CategoriesLayout>
                    </View>
                ))}
            </View >
        </ScrollView>

    </>
);


export default CategoriesRow;