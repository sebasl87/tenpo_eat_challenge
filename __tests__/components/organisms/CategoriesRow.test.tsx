import "react-native";
import React from "react";
import renderer from "react-test-renderer";

import { CategoriesRow } from "../../../src/components/organisms";

jest.mock("react-redux", () => ({
    ...jest.requireActual("react-redux"),
    useSelector: jest.fn()
        .mockReturnValueOnce({
            linksData: [
                {
                    uri: require("../../../assets/imgs/hamburguer.png"),
                    category: "HAMBURGUESAS"
                }]
        })
}));

describe("CategoriesRow component", () => {
    it("should render successfully with mock", () => {
        const categoriesRow = renderer
            .create(<CategoriesRow />)
            .toJSON();
        expect(categoriesRow).toMatchSnapshot();
    });
});
