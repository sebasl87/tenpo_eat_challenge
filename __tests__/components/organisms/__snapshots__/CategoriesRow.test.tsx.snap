// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesRow component should render successfully with mock 1`] = `
[
  <Text
    style={
      [
        {
          "color": "#333333",
          "fontSize": 18,
          "marginBottom": 6,
          "marginTop": 32,
        },
        {
          "fontFamily": "Gotham-Bold",
        },
      ]
    }
  >
    CATEGORIAS
  </Text>,
  <RCTScrollView
    horizontal={true}
  >
    <View>
      <View
        style={
          {
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            {
              "marginRight": 6,
            }
          }
        >
          <View
            style={
              [
                {
                  "alignContent": "center",
                  "alignItems": "center",
                  "display": "flex",
                  "height": 90,
                  "width": 168,
                },
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#fff",
                    "fontSize": 16,
                    "letterSpacing": 1,
                    "marginTop": 30,
                    "position": "absolute",
                    "zIndex": 2,
                  },
                  {
                    "fontFamily": "Gotham-Bold",
                  },
                ]
              }
            >
              HAMBURGUESAS
            </Text>
            <View
              accessibilityIgnoresInvertColors={true}
              style={
                [
                  {
                    "height": 80,
                    "width": 174,
                  },
                ]
              }
            >
              <Image
                source={
                  {
                    "testUri": "../../../assets/imgs/hamburguer.png",
                  }
                }
                style={
                  [
                    {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    {
                      "height": 80,
                      "width": 174,
                    },
                    undefined,
                  ]
                }
              />
            </View>
          </View>
        </View>
        <View
          style={
            {
              "marginRight": 6,
            }
          }
        >
          <View
            style={
              [
                {
                  "alignContent": "center",
                  "alignItems": "center",
                  "display": "flex",
                  "height": 90,
                  "width": 168,
                },
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#fff",
                    "fontSize": 16,
                    "letterSpacing": 1,
                    "marginTop": 30,
                    "position": "absolute",
                    "zIndex": 2,
                  },
                  {
                    "fontFamily": "Gotham-Bold",
                  },
                ]
              }
            >
              ITALIANA
            </Text>
            <View
              accessibilityIgnoresInvertColors={true}
              style={
                [
                  {
                    "height": 80,
                    "width": 174,
                  },
                ]
              }
            >
              <Image
                source={
                  {
                    "testUri": "../../../assets/imgs/italian.png",
                  }
                }
                style={
                  [
                    {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    {
                      "height": 80,
                      "width": 174,
                    },
                    undefined,
                  ]
                }
              />
            </View>
          </View>
        </View>
        <View
          style={
            {
              "marginRight": 6,
            }
          }
        >
          <View
            style={
              [
                {
                  "alignContent": "center",
                  "alignItems": "center",
                  "display": "flex",
                  "height": 90,
                  "width": 168,
                },
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#fff",
                    "fontSize": 16,
                    "letterSpacing": 1,
                    "marginTop": 30,
                    "position": "absolute",
                    "zIndex": 2,
                  },
                  {
                    "fontFamily": "Gotham-Bold",
                  },
                ]
              }
            >
              PIZZAS
            </Text>
            <View
              accessibilityIgnoresInvertColors={true}
              style={
                [
                  {
                    "height": 80,
                    "width": 174,
                  },
                ]
              }
            >
              <Image
                source={
                  {
                    "testUri": "../../../assets/imgs/pizza.png",
                  }
                }
                style={
                  [
                    {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    {
                      "height": 80,
                      "width": 174,
                    },
                    undefined,
                  ]
                }
              />
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>,
]
`;
